### PR TITLE
Ensure isa object method still works after using isa infix operator

### LINKED ISSUE #18679: isa experimental feature breaks isa object method
**Description**
When using both the isa infix operator and the isa object method, the isa object method no longer works.

**Steps to Reproduce**
```perl
use strict;
use warnings;

use feature 'say';

use experimental 'isa';

package Test;

sub new {
    my ($class) = @_;
    return bless {}, $class;
}

package main;

my $x = Test->new;

say $x isa 'Test';
say $x->isa('Test');
```

**Expected behavior**
Should print

```
1
1
```

Instead, it prints

```
1
Can't locate object method "isa" via package "Test" at test.pl line 20.
```

**Perl configuration**
```
Summary of my perl5 (revision 5 version 32 subversion 0) configuration:
   
  Platform:
    osname=linux
    osvers=3.10.0-1127.13.1.el7.x86_64
    archname=x86_64-linux
    uname='linux vm-mreisner.epic.com 3.10.0-1127.13.1.el7.x86_64 #1 smp tue jun 23 15:46:38 utc 2020 x86_64 x86_64 x86_64 gnulinux '
    config_args='-de -Dprefix=/home/mreisner/perl5/perlbrew/perls/perl-5.32.0 -Aeval:scriptdir=/home/mreisner/perl5/perlbrew/perls/perl-5.32.0/bin'
    hint=recommended
    useposix=true
    d_sigaction=define
    useithreads=undef
    usemultiplicity=undef
    use64bitint=define
    use64bitall=define
    uselongdouble=undef
    usemymalloc=n
    default_inc_excludes_dot=define
    bincompat5005=undef
  Compiler:
    cc='cc'
    ccflags ='-fwrapv -fno-strict-aliasing -pipe -fstack-protector-strong -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -D_FORTIFY_SOURCE=2'
    optimize='-O2'
    cppflags='-fwrapv -fno-strict-aliasing -pipe -fstack-protector-strong -I/usr/local/include'
    ccversion=''
    gccversion='4.8.5 20150623 (Red Hat 4.8.5-39)'
    gccosandvers=''
    intsize=4
    longsize=8
    ptrsize=8
    doublesize=8
    byteorder=12345678
    doublekind=3
    d_longlong=define
    longlongsize=8
    d_longdbl=define
    longdblsize=16
    longdblkind=3
    ivtype='long'
    ivsize=8
    nvtype='double'
    nvsize=8
    Off_t='off_t'
    lseeksize=8
    alignbytes=8
    prototype=define
  Linker and Libraries:
    ld='cc'
    ldflags =' -fstack-protector-strong -L/usr/local/lib'
    libpth=/usr/local/lib /usr/lib /lib/../lib64 /usr/lib/../lib64 /lib /lib64 /usr/lib64 /usr/local/lib64
    libs=-lpthread -lnsl -lgdbm -ldb -ldl -lm -lcrypt -lutil -lc -lgdbm_compat
    perllibs=-lpthread -lnsl -ldl -lm -lcrypt -lutil -lc
    libc=libc-2.17.so
    so=so
    useshrplib=false
    libperl=libperl.a
    gnulibc_version='2.17'
  Dynamic Linking:
    dlsrc=dl_dlopen.xs
    dlext=so
    d_dlsymun=undef
    ccdlflags='-Wl,-E'
    cccdlflags='-fPIC'
    lddlflags='-shared -O2 -L/usr/local/lib -fstack-protector-strong'


Characteristics of this binary (from libperl): 
  Compile-time options:
    HAS_TIMES
    PERLIO_LAYERS
    PERL_COPY_ON_WRITE
    PERL_DONT_CREATE_GVSV
    PERL_MALLOC_WRAP
    PERL_OP_PARENT
    PERL_PRESERVE_IVUV
    USE_64_BIT_ALL
    USE_64_BIT_INT
    USE_LARGE_FILES
    USE_LOCALE
    USE_LOCALE_COLLATE
    USE_LOCALE_CTYPE
    USE_LOCALE_NUMERIC
    USE_LOCALE_TIME
    USE_PERLIO
    USE_PERL_ATOF
  Built under linux
  Compiled at Dec 21 2020 10:01:36
  %ENV:
    PERLBREW_HOME="/home/mreisner/.perlbrew"
    PERLBREW_ROOT="/home/mreisner/perl5/perlbrew"
    PERLBREW_SHELLRC_VERSION="0.89"
  @INC:
    /home/mreisner/perl5/perlbrew/perls/perl-5.32.0/lib/site_perl/5.32.0/x86_64-linux
    /home/mreisner/perl5/perlbrew/perls/perl-5.32.0/lib/site_perl/5.32.0
    /home/mreisner/perl5/perlbrew/perls/perl-5.32.0/lib/5.32.0/x86_64-linux
    /home/mreisner/perl5/perlbrew/perls/perl-5.32.0/lib/5.32.0
```

### PR DESCRIPTION
**Pull request by FractalBoy**
Fixes #18679 

Added some tests to ensure isa object method still works after using isa infix operator.

All isa tests pass.

### CODE REVIEW

File: t/op/isa.t [hunk 1]
```diff
@@ -11,7 +11,7 @@ use strict;
 use feature 'isa';
 no warnings 'experimental::isa';
 
-plan 11;
```
Comment (Leont) [inline t/op/isa.t:14]: Can you please give your commits more descriptive titles? «Fix github issue 18679» doesn't tell a casual reader what it happening here, «Ensure isa object method still works after using isa infix operator» does (though mentioning github issue 18679 in the description would still be useful)

File: AUTHORS [hunk 2]
```diff
@@ -800,6 +800,7 @@ Manuel Valente                 <mvalente@idealx.com>
 Marc Green                     <marcgreen@cpan.org>
 Marc Lehmann                   <pcg@goof.com>
 Marc Paquette                  <Marc.Paquette@Softimage.COM>
+Marc Reisner                   <reisner.marc@gmail.com>
```
Comment (Leont) [inline AUTHORS:803]: This commit would be better described as «Add Marc Reisner to AUTHORS file»

**Comment by FractalBoy** [comment]
@Leont addressed your comments

File: universal.c [hunk 3]
```diff
@@ -216,12 +216,7 @@ Perl_sv_isa_sv(pTHX_ SV *sv, SV *namesv)
     if(!SvROK(sv) || !SvOBJECT(SvRV(sv)))
         return FALSE;
 
-    /* This abuse of gv_fetchmeth_pv() with level = 1 skips the UNIVERSAL
-     * lookup
-     * TODO: Consider if we want a NOUNIVERSAL flag for requesting this in a
```
Comment (leonerd) [inline universal.c:221]: Unless the `-1` part here really is relating to that NOUNIVERSAL behaviour, please leave this TODO comment in. It's a reminder to me that we still want that behaviour.

**Comment by leonerd** [comment]
The new tests seem a good thing to check.

Before I approve it I would like to check that the new code still correctly skips a call to the actual `UNIVERSAL::isa`, as my previous code did for purely efficiency reasons. There's no point doing a method lookup on the `->isa` method only to find it's the basic one from UNIVERSAL, when the `else` side of the `if` condition already implements the same behaviour, just more efficiently.

File: universal.c (hunk 3 above)
Comment (FractalBoy) [reply universal.c:221]: I thought the entire comment was one idea - I'll put back the TODO part of it.

Comment (FractalBoy) [reply universal.c:221]: Done - in commit ec8dc484c1

**Comment by FractalBoy** [comment]
>Before I approve it I would like to check that the new code still correctly skips a call to the actual `UNIVERSAL::isa`, as my previous code did for purely efficiency reasons. There's no point doing a method lookup on the `->isa` method only to find it's the basic one from UNIVERSAL, when the `else` side of the `if` condition already implements the same behaviour, just more efficiently.

@leonerd which `if`/`else` are you referring to? I don't understand how you can skip the call to `UNIVERSAL::isa` and still get the functionality from it? Why should infix `isa` function differently from `->isa` in this regard?

**Comment by FractalBoy** [comment]
Ah, I see `sv_derived_from_sv` should be called instead of falling back to `XS_UNIVERSAL_isa`.

**Comment by leonerd** [comment]
Ahah, a new `GV_NOUNIVERSAL` flag. That sounds like a proper fix for my TODO comment.

Having added the flag though I wonder if it needs extra documentation and so on now...

**Comment by FractalBoy** [comment]
@leonerd Documentation updated

**Comment by leonerd** [comment]
That looks good now. Waiting to see what the CI tests make of it...

**Comment by leonerd** [comment]
LGTM

Cygwin smoker is upset but then it's always upset, so I've taken to ignoring it.

Thoughts @Leont ?

**Comment by Leont** [comment]
LGTM

**Pull request merged**